### PR TITLE
Fix SAMRAIGhostDataAccumulator + SideData.

### PR DESCRIPTION
**Pull request by drwells**
Fixes one of the two remaining issues with #896.

The box returned by `getGhostBox` is for the patch: it is not the offset box required to correctly use side data. Using the wrong box caused the function to skip the last column which results in wrong accumulation values.

In the modified test output, note that 6 really should be the maximum value since that entry shows up four times on the patch interior (twice on each patch due to periodicity) and twice more on one patch in ghost entries.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by boyceg** [comment]
Are we using the SAMRAI geometry classes to generate appropriate boxes? E.g. [`SideGeometry<DIM>::toSideBox()`](https://ibamr.github.io/IBAMR-docs/samrai/2.4.4/html/classSAMRAI_1_1pdat_1_1SideGeometry.html#b84ba95e9888e7ffe365795f43ca2908)?

**Comment by drwells** [comment]
We should have, but we did not. This patch uses a much dumber solution - we just read everything in the arrays regardless of what the box actually is.

**Comment by drwells** [comment]
I'm not sure I understand your question. Are you asking about whether we should be using boxes somewhere else?

**Comment by boyceg** [comment]
> I'm not sure I understand your question. Are you asking about whether we should be using boxes somewhere else?

Sorry for not being clear --- probably the easiest thing to do is to rely on the boxes provided by the `ArrayData`. It looks like that is what you are doing?

However, if you want to know the range of indices that corresponds to the "interior" for a given data centering, one way to do that is to use the corresponding `Geometry` class.

**Comment by drwells** [comment]
The point of this patch is that we don't need boxes at all - we need all the data on the patch (interior and ghost are treated the same way for the purpose of accumulation) so we just read the underlying array. The actual box is irrelevant.

File: ibtk/src/math/SAMRAIGhostDataAccumulator.cpp [hunk 1]
```diff
@@ -124,6 +111,8 @@ add_or_get(Pointer<PatchLevel<NDIM> >& level,
         IBTK_CHKERRQ(ierr);
         for (unsigned int d = 0; d < dofs_ptrs.size(); ++d)
         {
+            const PetscInt size = dofs_ptrs[d]->getDepth() * dofs_ptrs[d]->getOffset();
```
Comment (drwells) [inline ibtk/src/math/SAMRAIGhostDataAccumulator.cpp]: This is the part that replaces the box usage - `getDepth() * getOffset()` is exactly the size of the array.

**Pull request merged**